### DISCUSSION TITLE
Webhook support

### DISCUSSION THREAD

**Discussion by jibon57** [mynaparrot/plugNmeet-server#3] [Announcements]
From server version 1.0.1 webhook support added. 

- Similar way livekit is sending webhook
- Use `HTTP_AUTHORIZATION` header with hash
- Body type `JSON`

Example in PHP:
```
// to get header
$hash = $_SERVER["HTTP_AUTHORIZATION"];

// to get body
$entityBody = file_get_contents('php://input');

```

Add webhook URL:

-  You can set global url in config.yaml file under `webhook_conf`. Make sure value of `enable` is true.
- During create room you can send custom webhook url too. To enable this require `enable_for_per_meeting` to `true`

You can find example: https://github.com/mynaparrot/plugNmeet-sdk-php/blob/main/examples/webhook.php

**Comment by shubhendumadhukar** [comment]
Is there a documentation available for webhooks? List of events, data schema etc

**Reply by jibon57** [reply]
You can get idea from here: https://github.com/mynaparrot/plugnmeet-protocol/blob/a61c90132af974d82d9cab0fd25cb4964f1d4c64/proto_files/plugnmeet_common.proto#L8
We don't have any documentation yet but contributors are welcome.

**Comment by hadyk77** [comment]
@jibon57 
I'm trying to enable webhook and receive session events here is the configuration of the webhook section in config.yaml 

![Screenshot 2024-06-27 at 3 00 29 PM](https://github.com/mynaparrot/plugNmeet-server/assets/36902888/444dcd3f-34e9-4501-88ca-6e9ae2909e85)

is there another place need to edit either?

**Reply by jibon57** [reply]
You can always check logs to find out problems.
